### PR TITLE
Added Record Screen command

### PR DESCRIPTION
**Pull request by janostik**
## Description

Comands which simplifies the start of new screen recording via QuickTime Player. 

## Type of change

- New extension

## Screencast

Command Preview

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I checked that extension's icon has size of `512x512`

### CODE REVIEW

**Comment by pernielsentikaer** [comment]
@janostik is there anything I can help you with?

**Comment by unnamedd** [comment]
> Wouldn't it make more sense to add this as a script command instead?

Per, did you have some answer for that? This extension definitely could be easily done with Script Command instead of Extension, and I am really interested to know why @janostik consider it better as extension.

**Comment by pernielsentikaer** [comment]
> This extension definitely could be easily done with Script Command instead of Extension, and I am really interested to know why @janostik consider it better as extension.

Having it as an extension have one major benefit compared to script commands, they are easily shared and installable by others which already have a lot more potential having it this way.

**Comment by PitNikola** [comment]
Our official stand: there is nothing wrong with having extensions in the store that could've been scripts.
The only exception would be if functionality doesn't leverage rich UI where needed.

**Comment by janostik** [comment]
@pernielsentikaer Sorry for late response from my end, but I was offline for couple of days. If by scripts you mean standard script in terminal and not some feature in raycast I'm not aware of, my reasoning is following. 

When I prepare scripts that I need to use, but do not use frequently (convert video to gif, start google meets quickly, ... ) I've found that it's easier for me to register these in raycast. Instead of opening the terminal and trying to remember if I've created the script and how I named it, I can open raycast and do fulltext search for action I need. Also I'm not sure if the targeted audience of the raycast app are people opening terminal.

Why this command specifically, it's because I've noticed that people around me using macs, have no idea there is a way you can record the screen or there might some specific shortcut do so. 

But I'm not against removing this PR if you think this could be replaced by something more sophisticated like connection to specific app like CleanShot.

**Comment by unnamedd** [comment]
> Having it as an extension have one major benefit compared to script commands, they are easily shared and installable by others which already have a lot more potential having it this way.

Sorry Per, but I have to disagree with you. It is amazing that Extensions has an easier way to be installed, but the Script Commands are also easily shareable, hence the Awesome Script Command available in the repository. One doesn’t exclude the other.

My understanding about the Extensions is that it is better used when you need UI, something that isn’t provided by the Script Commands.

**Comment by PitNikola** [comment]
@unnamedd, I think Per meant that distributing and installing extensions is easier through the store comparing to script commands through script-commands repo. We received a lot of feedback from users who are not familiar with GitHub that they struggled importing scripts into Raycast.

Extensions make it easier to group commands, they can be easily updated, all the resources are bundled inside, etc. For this reason we won't reject extensions that might easily be script-commands. In fact this [Spotify Controls](https://www.raycast.com/thomas/spotify-controls) is a port of Spotify Control scripts.

Having said that scripts are simpler to get started with from developer side, so for now they can co-exist with extensions as just simpler workflows.

**Comment by unnamedd** [comment]
@PitNikola, I am not disagreeing that Script Commands is a bit harder to keep under a good management, but the problem solved for the Extensions was possible due to the lessons learned with Script Commands, doesn’t mean that now the Script Commands are evil because they are not so simple to be managed. This is something that isn’t really easy to solve but it’s also not very hard. 

Definitely, we should find a better way to solve the Script Commands management.

but that’s just my 2 cents in the discussion.

**Comment by PitNikola** [comment]
Nobody said that it's evil or something :)

File: extensions/screen-recording/package.json [hunk 1]
```diff
@@ -0,0 +1,35 @@
+{
+  "name": "screen-recording",
+  "title": "Screen Recording",
```
Comment (PitNikola) [inline extensions/screen-recording/package.json:3]:
```suggestion
  "title": "Quicktime Screen Recording",
```

File: extensions/screen-recording/package.json [hunk 2]
```diff
@@ -0,0 +1,35 @@
+{
+  "name": "screen-recording",
```
Comment (PitNikola) [inline extensions/screen-recording/package.json:2]:
```suggestion
  "name": "quicktime-screen-recording",
```

**Review by PitNikola — CHANGES_REQUESTED** [review]
@janostik, I've only got minor suggestions on the naming and then we are good to go.

You might also want to rename the folder to `quicktime-screen-recording`

**Comment by pernielsentikaer** [comment]
@janostik do you need any help?

**Comment by fe9lix** [comment]
Closing this PR due to inactivity – feel free to reopen later.

**Pull request closed without merging**